src/app/item/item-list/item-list.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component } from '@angular/core';
+import { GroceryService } from 'src/app/grocerservice.service';
+//import { Item } from '../models/item';
+import { Item } from 'src/app/models/item';
+
+@Component({
+  selector: 'app-item-list',
+  templateUrl: './item-list.component.html',
+  styleUrls: ['./item-list.component.css'],
+})
+export class ItemListComponent {
+  items: Promise<Item[]> = this.groceryService.items;
+  filteredItems: Promise<Item[]> = this.groceryService.items;
+  types: string[] = ['vegetable', 'fruit', 'all'];
+  index: number = 0;
+
+  constructor(private readonly groceryService: GroceryService) {
+    this.items = this.groceryService.items;
+    console.log(this.items);
+    this.filteredItems = this.groceryService.items;
+    console.log(this.items);
+  }
+
+  addToCart(item: Item) {
+    this.groceryService.addToCart(item);
+  }
+
+  filterType(value: object) {
```

Comment:
Your filtering method filterType is reliant on an index to cycle through types. While this works, it could lead to potential bugs if the types array changes. Consider directly passing the type to filter as a parameter to the filterType method, which makes the function more predictable and easier to understand

```typescript
filterType(selectedType: string) {
  if (selectedType === 'all') {
    this.filteredItems = this.items;
  } else {
    this.filteredItems = this.items.then(items =>
      items.filter(item => item.type === selectedType)
    );
  }
}

```